src/main/java/uk/gov/hmcts/cmc/claimstore/services/IntentionToProceedService.java

```diff
@@ -1,26 +1,134 @@
+import java.util.Collections;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+import static uk.gov.hmcts.cmc.claimstore.appinsights.AppInsights.REFERENCE_NUMBER;
 
 @Service
 public class IntentionToProceedService {
-    private final int intentionToProceedDeadline;
+
+    Logger logger = LoggerFactory.getLogger(this.getClass());
 
     private final WorkingDayIndicator workingDayIndicator;
 
+    private final CaseSearchApi caseSearchApi;
+
+    private final UserService userService;
+
+    private final CaseRepository caseRepository;
+
+    private final AppInsights appInsights;
+
+    private final IntentionToProceedContentProvider emailContentProvider;
+
+    private final EmailService emailService;
+
+    private final StaffEmailProperties emailProperties;
+
+    private final IntentionToProceedDeadlineCalculator intentionToProceedDeadlineCalculator;
+
     public IntentionToProceedService(
         WorkingDayIndicator workingDayIndicator,
-        @Value("${intention.to.proceed.deadline:33}") int intentionToProceedDeadline
+        CaseSearchApi caseSearchApi,
+        UserService userService,
+        AppInsights appInsights,
+        CaseRepository caseRepository,
+        IntentionToProceedContentProvider emailContentProvider,
+        EmailService emailService,
+        StaffEmailProperties emailProperties,
+        IntentionToProceedDeadlineCalculator intentionToProceedDeadlineCalculator
     ) {
         this.workingDayIndicator = workingDayIndicator;
-        this.intentionToProceedDeadline = intentionToProceedDeadline;
+        this.caseSearchApi = caseSearchApi;
+        this.userService = userService;
+        this.caseRepository = caseRepository;
+        this.appInsights = appInsights;
+        this.emailContentProvider = emailContentProvider;
+        this.emailService = emailService;
+        this.emailProperties = emailProperties;
+        this.intentionToProceedDeadlineCalculator = intentionToProceedDeadlineCalculator;
     }
 
-    public LocalDate calculateIntentionToProceedDeadline(LocalDate respondedAt) {
-        LocalDate intentionToProceedDeadline = respondedAt.plusDays(this.intentionToProceedDeadline);
-        return workingDayIndicator.getNextWorkingDay(intentionToProceedDeadline);
+    @Scheduled(cron = "#{'${claim_stayed.schedule}' ?: '-'}")
+    public void scheduledTrigger() {
+        LocalDateTime now = LocalDateTime.now();
+        if (workingDayIndicator.isWorkingDay(now.toLocalDate())) {
+            User anonymousCaseWorker = userService.authenticateAnonymousCaseWorker();
+            checkClaimsPastIntentionToProceedDeadline(now, anonymousCaseWorker);
+        }
+    }
+
+    public void checkClaimsPastIntentionToProceedDeadline(LocalDateTime dateTime, User user) {
+
+        //4pm cut off for court working days
+        int adjustDays = dateTime.getHour() >= 16 ? 0 : 1;
+        LocalDate runDate = dateTime.toLocalDate().minusDays(adjustDays);
```

Comment:
Do you think this would be better in the **intentionToProceedDeadlineCalculator.calculateResponseDate(runDate)** method itself as you will always need to make this check right?